### PR TITLE
figma-linux: add NIXOS_OZONE_WL workaround

### PR DESCRIPTION
**Pull request by mrnossiom**
## Description of changes

I recently took down `xwayland` from my config and noticed that `figma-linux` crashed because it could not find an X display. I did knew about the `NIXOS_OZONE_WL` fix but noticed it wasn't applied to this package. I copied this line from another Electron app and it seems to work fine.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@ercao @kashw2

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/graphics/figma-linux/default.nix [hunk 1]
```diff
@@ -17,7 +17,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-9UfyCqgsg9XAFyZ7V7TogkQou4x+ixFUfjXZ1/qlDmA=";
   };
 
-  nativeBuildInputs = [ autoPatchelfHook dpkg wrapGAppsHook3 ];
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
```
[the remark this answers is not in the document]
Comment (mrnossiom) [reply pkgs/applications/graphics/figma-linux/default.nix:23]: You mean above the others ? I sorted alphabetically

[the remark this answers is not in the document]
Comment (mrnossiom) [reply pkgs/applications/graphics/figma-linux/default.nix:23]: Oh right

**Review by O8888 — APPROVED** [review]

**Review by O8888 — APPROVED** [review]

File: pkgs/applications/graphics/figma-linux/default.nix [hunk 2]
```diff
@@ -17,7 +18,12 @@ stdenv.mkDerivation (finalAttrs: {
...
+    wrapGAppsHook3
```
Comment (Aleksanaa) [inline pkgs/applications/graphics/figma-linux/default.nix:25]: By adding `wrapGAppsHook3` and manually `wrapProgram`, you are wrapping `figma-linux` two times. See https://github.com/NixOS/nixpkgs/blob/nixos-unstable/doc/languages-frameworks/gnome.section.md#when-using-wrapgapps-hook-with-special-derivers-you-can-end-up-with-double-wrapped-binaries-ssec-gnome-common-issues-double-wrapped

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 309406` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>figma-linux</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]
Seems to run perfectly under wayland!

**Pull request merged**